### DISCUSSION TITLE
Quarto - Setting up a "book" as a single notebook post?

### DISCUSSION THREAD

**Discussion by kubu4** [RobertsLab/resources#1719] [Q&A]
I want to have an R Project rendered as a single blog post, but the issue is that we generally organize our R Projects in the following fashion:

```
├── ceabigr.Rproj
├── code
│   ├── 02-te-all.sh
│   ├── 07-TE-intersect-meth.Rmd
│   ├── 10-TEm-TEcn.Rmd
│   ├── 11.1-haplotype.sh
│   ├── 11.2-haplotype.sh
│   ├── 11-RNAseq-snps.Rmd
│   ├── 12-lncRNA.Rmd
│   ├── 13-lncRNA-kallisto.Rmd
│   ├── 14-RNA-kallisto.Rmd
│   ├── 15-RNAseq.Rmd
│   ├── 16-DML-annotation.Rmd
│   ├── 17-Swiss-Prot-Annotation.Rmd
├── output
│   ├── 02-te-all
│   ├── 11-RNAseq-snps
│   ├── 12-lncRNA
│   ├── 13-lncRNA-kallisto
│   ├── 14-RNA-kallisto
│   ├── 15-Gene-expression-analysis
│   ├── 16-DML-annotation
│   ├── 17-Swiss-Prot-Annotation
```

Each Rmd file has a corresponding output directory of the same name.

So, if I followed this format (which I'd like to do), I'd have a `.qmd` file (instead of `.Rmd` file) for each type of analysis I'd run.

In the current usage of Quarto for our notebooks, each `.qmd` file would get rendered as separate notebook posts.

However, I'd like a single notebook post, but with the ability to go through all the analyses performed in the R Project.

I _think_ the solution is to set up the R Project as a Quarto Book. This allows defining "chapters" and the resulting rendered HTML files show a table of contents across the multiple `.qmd` files. 

Unfortunately, it seems like when the Quarto Book is rendered in the notebook, each individual `.qmd` file gets rendered as it's own notebook entry.

Anyone have any thoughts on how to get a Quarto Book to function as a single notebook entry?

**Comment by sr320** [comment]
Compiling multiple R Markdown or Quarto documents into a single notebook entry or blog post while retaining a structured and organized file system can be a bit tricky, but it's definitely doable with some workarounds. Here are some suggestions you might find helpful:

1. **Single Master Document**:
    - You can create a single master `.Rmd` or `.qmd` file that serves as a wrapper around all your individual analysis documents. 
    - You can use the `child` option in the YAML header to include the content of the other R Markdown files.
    - Example:

```
        ```yaml
        ---
        title: "Master Document"
        output: html_document
        ---

        ```{r child='code/07-TE-intersect-meth.Rmd'}
        ```

        ```{r child='code/10-TEm-TEcn.Rmd'}
        ```
        ```
```

This way, when you knit the master document, all the child documents get rendered into a single HTML document.

2. **Bookdown**:
    - You can use `bookdown` instead of Quarto, which is designed to handle multi-document projects and can compile them into a single output.
    - Create an `_bookdown.yml` file to specify the order of the documents.
    - Example:
        ```yaml
        rmd_files:
          - code/07-TE-intersect-meth.Rmd
          - code/10-TEm-TEcn.Rmd
          ...
        ```
    - Then, render the book with `bookdown::render_book()` to get a single HTML file with a table of contents.

3. **Custom Script**:
    - Write a custom script that programmatically concatenates your `.qmd` or `.Rmd` files into a single document before rendering.
    - This can be done with basic file I/O operations in R or another scripting language.

4. **External Tools**:
    - There are also external tools and platforms like Distill for R, which might provide more seamless integration for combining multiple R analyses into a single blog post or notebook entry.

5. **Custom CSS/JS**:
    - If the issue is more about the aesthetics or navigation, you might be able to solve the problem with custom CSS or JavaScript to create a single-page feel, or a navigable table of contents across multiple HTML documents.

6. **Quarto Book**:
    - Explore further the Quarto documentation or reach out to the Quarto community for support. There might be undocumented or advanced features that can address this specific use case.

7. **Explore Other Tools**:
    - If none of the above solutions work, you might want to explore other documentation or blog rendering tools that better fit your project structure and rendering needs.

**Reply by kubu4** [reply]
Interesting. Option 1 seems like the most straightforward way. I'll test it out and report back. Thanks!

When I asked ChatGPT, it didn't provide me with all of these options...
